frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx

```diff
@@ -0,0 +1,376 @@
+          path: ['dateOfBirth'],
+        });
+      } else if (!isPast(parsedDateOfBirth)) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: t('apply:child-information.error-message.date-of-birth-is-past'),
+          path: ['dateOfBirth'],
+        });
+      } else if (differenceInYears(new Date(), parsedDateOfBirth) > 150) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: t('apply:child-information.error-message.date-of-birth-is-past-valid'),
+          path: ['dateOfBirth'],
+        });
+      }
+    })
+    .transform((val) => {
+      // At this point the year, month and day should have been validated as positive integer
+      const parseDateOfBirthString = parseDateString(`${val.dateOfBirthYear}-${val.dateOfBirthMonth}-${val.dateOfBirthDay}`);
+      return {
+        ...val,
+        dateOfBirth: `${parseDateOfBirthString.year}-${parseDateOfBirthString.month}-${parseDateOfBirthString.day}`,
+      };
+    }) satisfies z.ZodType<ChildInformationState>;
+
+  const data = {
+    firstName: String(formData.get('firstName') ?? ''),
+    lastName: String(formData.get('lastName') ?? ''),
+    dateOfBirthYear: formData.get('dateOfBirthYear') ? Number(formData.get('dateOfBirthYear')) : undefined,
+    dateOfBirthMonth: formData.get('dateOfBirthMonth') ? Number(formData.get('dateOfBirthMonth')) : undefined,
+    dateOfBirthDay: formData.get('dateOfBirthDay') ? Number(formData.get('dateOfBirthDay')) : undefined,
+    dateOfBirth: '',
+    hasSocialInsuranceNumber: String(formData.get('hasSocialInsuranceNumber') ?? ''),
+    socialInsuranceNumber: formData.get('socialInsuranceNumber') ? String(formData.get('socialInsuranceNumber') ?? '') : undefined,
+    isParent: String(formData.get('isParent') ?? ''),
+  };
+
+  const parsedDataResult = childInformationSchema.safeParse(data);
+  if (!parsedDataResult.success) {
+    return json({ errors: parsedDataResult.error.format() });
+  }
+
+  await saveApplyAdultState({ params, request, session, state: { childInformation: parsedDataResult.data } });
+
+  if (state.adultState.editMode) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/review-information', params));
+  }
+
+  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/personal-information', params));
+}
+
+export default function ApplyFlowChildInformation() {
+  const { t } = useTranslation(handle.i18nNamespaces);
+  const { csrfToken, defaultState, editMode } = useLoaderData<typeof loader>();
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
+  const isSubmitting = fetcher.state !== 'idle';
+  const errorSummaryId = 'error-summary';
+  const [hasSocialInsuranceNumberValue, setHasSocialInsuranceNumberValue] = useState(defaultState?.hasSocialInsuranceNumber ?? '');
+
+  const handleOnPreferredMethodChecked: ChangeEventHandler<HTMLInputElement> = (e) => {
+    setHasSocialInsuranceNumberValue(e.target.value);
+  };
```

Comment:
**Nitpick:** This function name can be misleading as we're dealing with SIN selection. Maybe something like `handleSocialInsuranceNumberSelection` (or similar) would be more appropriate?

Reply:
Guess you now know I copied and pasted from somewhere ;)